### DISCUSSION TITLE
How to add text background-color?

### DISCUSSION THREAD

**Discussion by dsantiago** [typst/typst#845] [Q&A] [answered]
I used the #block tag but it adjust to full line, how to have background-color only for a specific phrase?

**Comment by PgBiel** [comment]
Use `#box` instead, e.g.
```
Text #box(fill: red)[this is red]this is not
```

**Comment by dsantiago** [comment]
Tried: `#box(fill: red)[#lorem(30)]`

<img width="606" alt="image" src="https://user-images.githubusercontent.com/3484029/232368345-e9fe9377-c639-4cd9-8397-f15a25cde051.png">

Too bad it's a box, it's not related to the text and does not stop at the text end.

**Reply by PgBiel** [reply, marked as answer]
Oh, I see what you mean - you want background that adjusts to the text. Right now this isn't natively supported, but here's a rough hack:
```
#[
   #show regex("\\w+|[^\\s]"): box.with(fill: red)
   #show " ": box.with(fill: red, width: 2pt)
   #lorem(30)
]
```
(Places every word - or, otherwise, character - in an individually filled box, with a special case for space (a box with just a space and auto width is automatically collapsed))

This should get better in the future - please open a feature request issue for this.

**Comment by dsantiago** [comment]
Tried it now! The result is the expected but text changes all spacing. The better approximation was `width: 2.7pt` but it's still visible words mixing up. Thanks anyway.

```rust
#[
   #show regex(".*"): box.with(fill: red)
   #show " ": box.with(fill: red, width: 2.7pt)
   #lorem(30)
]
```

**Comment by dsantiago** [comment]
Just to note, found the point why the text changes so much, my `text` property is set with `hyphenate: true` and with box property the words are justified together ofc.

For further tests use:

```rust
set text(size: 12pt, hyphenate: true)
#[
   #show regex(".*"): box.with(fill: red)
   #show " ": box.with(fill: red, width: 2.7pt)
   #lorem(30)
]
```

**Reply by PgBiel** [reply]
Ah; I guess an inner `set text(hyphenate: false)` inside the desired content block (`#[ ]`) will be a good enough fix (that doesn't require changing the property outside).

**Reply by dsantiago** [reply]
Yeah, in this case I must respect the pattern... By the way, obrigado amigo!

**Reply by PgBiel** [reply]
Gotcha. De nada, camarada!

**Reply by SylvanFranklin** [reply]
Have there been any developments here? I'm still looking at how to do this

**Reply by PgBiel** [reply]
Yes, this is now built-in: https://typst.app/docs/reference/text/highlight/

**Reply by SylvanFranklin** [reply]
Should this be marked as closed or is it important as a piece of documentation to searchers?

**Reply by PgBiel** [reply]
It is already closed! Though, if you have further questions, we encourage you to try out our forum: https://forum.typst.app :)
